### DISCUSSION TITLE
Add test for tutorial notebooks

### DISCUSSION THREAD

**Discussion by NimaSarajpoor** [TDAmeritrade/stumpy#827] [General]
I was wondering if we could add a test for tutorial notebooks. Even if we do not put them in the CI flow, we may still get beenfit from this test as developers can run it locally and see if their changes result in any issue in the tutorial notebooks. This can become important if commits change code in `core.py` or `stump.py`, which may affect more than one notebook.

Can we do something like this: https://github.com/TDAmeritrade/stumpy/pull/826/files ? It is not complete as we need to remove some of the generated files (or maybe use an option to do that) and we need to continue testing notebooks even if one of them fails rather than being stopped at the first occurance of an error.

---
Also:

It seems that the following notebooks cannot be run successfully: 
```
docs/Tutorial_DiscordMERLIN.ipynb
docs/Tutorial_MPdist.ipynb
docs/Tutorial_meter_swapping.ipynb
```

In addtion, the notebook `docs/Tutorial_Matrix_Profiles_For_Streaming_Data.ipynb` takes too much time. Maybe this is because my hardware is old (I am using a 10-year old pc, with 2 cores!) So, it would be great if it can be run on a newer device. If the running time issue persists, then I guess the reason is probably because of the fact that the existing version computes the sliding mean / std in a non-welford manner which is slower than welford approach. 

Note: In one of the cells, we run `stumpy.stump` several times (approx. `10,000` iterations) as we update `T`, and the purpose is to show how its running time is compared to the performance of `stumpy/stumpi.py`.

**Comment by seanlaw** [comment]
@NimaSarajpoor Perhaps, we can add the tests as you've proposed but, instead of running it here (and competing for the same CI resources), we can execute the test in the `stumpy_dev` `automate` repository: https://github.com/stumpy-dev/automate

There, we can add a new workflow that gets triggered whenever there is a new commit or new PR and it can check tutorials.

> Note: In one of the cells, we run stumpy.stump several times (approx. 10,000 iterations) as we update T, and the purpose is to show how its running time is compared to the performance of stumpy/stumpi.py.

Yes, that was on purpose and it is expected to be slow

**Reply by NimaSarajpoor** [reply]
> There, we can add a new workflow that gets triggered whenever there is a new commit or new PR and it can check tutorials.

That sounds an interesting idea.



> Yes, that was on purpose and it is expected to be slow

Yes, it is expected to be slowER; however, it shouldn't take that much time for a tutorial. It is not a big deal. We can leave it as is. Later, if we notice that it affects the running time of test we are going to add for tutorial notebooks, then we can try to resolve it.

**Reply by seanlaw** [reply]
> Yes, it is expected to be slowER; however, it shouldn't take that much time for a tutorial. It is not a big deal. We can leave it as is. Later, if we notice that it affects the running time of test we are going to add for tutorial notebooks, then we can try to resolve it.

Honestly, I didn't expect people to run that particular tutorial. I only expected people to read it
